pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -706,8 +707,9 @@ public void start() throws PulsarServerException {
             this.startNamespaceService();
 
             schemaStorage = createAndStartSchemaStorage();
-            schemaRegistryService = SchemaRegistryService.create(
-                    schemaStorage, config.getSchemaRegistryCompatibilityCheckers());
+            ensureSchemaRegistryName(schemaStorage);
```

Comment:
`createAndStartSchemaStorage ` will create schemaStorage, if don't set the schemaRegistryStorageClassName, it will throw NPE, so this method seem not useful or need to add a default SchemaRegistryStorageClassName

Reply:
@congbobo184 the default schema registry name is being set in ServiceConfiguration so NPE would not happen here

Reply:
I mean why we need check schemaStorage == null in ensureSchemaRegistryName

Reply:
this was part of the existing logic. if schema storage is null then the SchemaRegistryServiceDisabled needs to be used. otherwise, SchemaRegistryServiceImpl needs to be used by default. 
below is the expected behavior for each of the combinations for configs set in broker's conf file.

* if schemaRegistryClassName == null and schemaRegistryStorageClassName == null then set schemaStorage to null and schemaRegistry to SchemaRegistryServiceDisabled
* if schemaRegistryClassName != null and schemaRegistryStorageClassName == null then set schemaStorage to null and schemaRegistry to SchemaRegistryServiceDisabled
* if schemaRegistryClassName == null and schemaRegistryStorageClassName != null then set schemaStorage to the configured object and schemaRegistry to SchemaRegistryServiceImpl
* if schemaRegistryClassName != null and schemaRegistryStorageClassName != null then set schemaStorage to the configured object and schemaRegistry to the configured object

Reply:
in the older code, this logic was implemented in `SchemaRegistryService.create(...)`

Reply:
https://github.com/apache/pulsar/blob/caf7648653c140c9f922e86425585ab7b5ed3ed6/pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java#L1303
in this code, createAndStartSchemaStorage will not create a schemaStorage is null. so why we should check ensureSchemaRegistryName and set default?

Reply:
you are right, schemaStorage can never be null. i had only kept this null check to retain the existing logic. i have removed it now.